### PR TITLE
feat(grid): add narrow grid mode

### LINKED ISSUE #6464: Narrow grid support
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Summary

In the Carbon documentation a "narrow" grid is mentioned (16px gutter grid):
https://www.carbondesignsystem.com/guidelines/2x-grid/implementation#the-narrow-grid

It shows a notification that the naming convention is not consistent and the demo shows a grid which has more padding on the right (end) than on the left:
image

**Is there any plan to make this grid mode official?**
I can't find any class which would indicate a "narrow grid" (only condensed gird). And the version on the site is not really usable (using `bx--no-gutter--left`), because of the stated spacing issue.

**What is the best way to implement this grid Carbon compliant?**

## Relevant information

My design team started to use this grid mode (16px) more often and I'm not sure what to advise them or the dev team (on how to implement it using `bx--grid`).

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #6464 
Ref #6510 

Add narrow grid mode
_*gutter classes will be updated in separate PR_

#### Changelog

**New**

- Adds new narrow grid mode classes for grid and row
- Adds narrow prop to grid and row components

#### Testing / Reviewing

- Elements preview - https://deploy-preview-6520--carbon-elements.netlify.app/grid/examples/preview/#narrow-grid
- Narrow grid mode story - https://deploy-preview-6520--carbon-components-react.netlify.app/?path=/story/grid--narrow

Make sure narrow grid is rendering correctly and nothing is broken. 
Narrow grid docs - https://www.carbondesignsystem.com/guidelines/2x-grid/implementation#the-narrow-grid

### CODE REVIEW

File: packages/grid/scss/_mixins.scss [hunk 1]
```diff
@@ -41,6 +41,13 @@
     padding-right: ($condensed-gutter / 2);
     padding-left: ($condensed-gutter / 2);
   }
+
+  // For our narrow use-case, our container hangs 16px into the gutter
+  .#{$prefix}--row--narrow &,
```
Comment (emyarod) [inline packages/grid/scss/_mixins.scss:46]:
```suggestion
  .#{$prefix}--row--narrow,
```

File: packages/grid/scss/_mixins.scss [hunk 2]
```diff
@@ -41,6 +41,13 @@
...
+  .#{$prefix}--grid--narrow & {
```
Comment (emyarod) [inline packages/grid/scss/_mixins.scss:47]:
```suggestion
  .#{$prefix}--grid--narrow {
```

**Review by emyarod — COMMENTED** [review]
just wondering, are the ampersands needed in the mixin selectors?

File: packages/grid/scss/_mixins.scss (hunk 1 above)
Comment (alisonjoseph) [reply packages/grid/scss/_mixins.scss:46]: Yes I believe so, I followed the same pattern as the condensed grid, these styles need to apply to the children of the class name.

**Review by emyarod — APPROVED** [review]
thanks for clarifying, looks good to me

File: packages/react/src/components/Grid/Row.js [hunk 3]
```diff
@@ -43,6 +45,12 @@ Row.propTypes = {
    */
   condensed: PropTypes.bool,
 
+  /**
+   * Specify a single row as narrow. The ontainer will hang
```
Comment (alisonjoseph) [inline packages/react/src/components/Grid/Row.js]:
```suggestion
   * Specify a single row as narrow. The container will hang
```

**Review by tw15egan — APPROVED** [review]
Tested locally and everything looks good 👍 ✅ 

Screen Shot 2020-07-23 at 11 06 17 AM

**Pull request merged**